mantidproject/mantid
# Fix add ranges in MultiFileNameParser

This PR enables handling of complex add ranges in `MultiFileNameParser` and consequently in `Load`, `LoadAndMerge` and similar algorithms that accept ranges of run numbers.

Previously, the parsing of an input string was rigidly based on regular expression matching which did not take into account more advanced expressions like `1+2-6+8-10+15`. Now, the tokens in the input string are further broken into smaller pieces. Only the individual pieces are matched against regexps which makes is possible to parse more complex patterns.


**Report to:** Jacques Ollivier ILL.

**To test:**

Try to `Load` or `LoadAndMerge` some files using different combinations of run number patterns such as `1+4-5+10-15:2` and check that the correct runs get loaded and merged.

Fixes #22597.

Does this update require release notes?

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
